src/ru/mangabuff/src/eu/kanade/tachiyomi/extension/ru/mangabuff/MangaBuff.kt

```diff
@@ -0,0 +1,343 @@
+                filter.checked?.forEach { addQueryParameter("status_id[]", it) }
+            }
+            (filters.find { it is AgeFilter } as? AgeFilter)?.let { filter ->
+                filter.checked?.forEach { addQueryParameter("age_rating[]", it) }
+            }
+            (filters.find { it is RatingFilter } as? RatingFilter)?.let { filter ->
+                filter.checked?.forEach { addQueryParameter("rating[]", it) }
+            }
+            (filters.find { it is YearFilter } as? YearFilter)?.let { filter ->
+                filter.checked?.forEach { addQueryParameter("year[]", it) }
+            }
+            (filters.find { it is ChapterCountFilter } as? ChapterCountFilter)?.let { filter ->
+                filter.checked?.forEach { addQueryParameter("chapters[]", it) }
+            }
+            (filters.find { it is GenreFilter } as? GenreFilter)?.let { filter ->
+                filter.excluded?.forEach { addQueryParameter("without_genres[]", it) }
+            }
+            (filters.find { it is TypeFilter } as? TypeFilter)?.let { filter ->
+                filter.excluded?.forEach { addQueryParameter("without_type_id[]", it) }
+            }
+            (filters.find { it is TagFilter } as? TagFilter)?.let { filter ->
+                filter.excluded?.forEach { addQueryParameter("without_tags[]", it) }
+            }
+            (filters.find { it is SortFilter } as? SortFilter)?.let { filter ->
+                addQueryParameter("sort", filter.selected)
+            }
+            if (page != 1) addQueryParameter("page", page.toString())
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaSelector() = ".cards .cards__item"
+
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        title = element.selectFirst(".cards__name")!!.text()
+
+        val slug = "$baseUrl$url".toHttpUrl().pathSegments.last()
+        thumbnail_url = "$baseUrl/img/manga/posters/$slug.jpg"
+    }
+
+    override fun searchMangaNextPageSelector() =
+        ".pagination .pagination__button--active + li:not(:last-child)"
+
+    // Details
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        title = document.selectFirst("h1, .manga__name, .manga-mobile__name")!!.text()
+
+        description = buildString {
+            document
+                .selectFirst(".manga__description")
+                ?.text()
+                ?.also { append(it) }
+
+            document // rating%
+                .selectFirst(".manga__rating")
+                ?.text()
+                ?.toDoubleOrNull()
+                ?.let { it / 10.0 }
+                ?.also {
+                    if (isNotEmpty()) append("\n\n")
+                    append(String.format(Locale("ru"), "Рейтинг: %.0f%%", it * 100))
```

Comment:
if string is empty, it would still add it the description no?

Reply:
It will still add the rating to the description, only the `\n\n` is conditional -- if the description is already empty it won't add newlines.

Might be easier to see here, which is equivalent:
```kt
if (isNotEmpty()) {
    append("\n\n")
}
append(String.format(Locale("ru"), "Рейтинг: %.0f%%", it * 100))
```

Should I change it so I explicitly use the curly braces, like the above code?

Reply:
oh ok nvm, for some reason I was under the impression that the check was for the rating being added.